resources/resource_protocol/config.html

```diff
@@ -0,0 +1,328 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <title>Advanced Config</title>
+    <style>
+        body {
+            padding: 0;
+            display: flex;
+            flex-direction: column;
+        }
+
+        input {
+            border-radius: 4px;
+        }
+
+        .search {
+            height: 24px;
+            width: calc(100% - 6px);
+            margin-top: 12px;
+            margin-bottom: 12px;
+            margin-right: 6px;
+            padding: 4px;
+        }
+
+        table {
+            border-collapse: collapse;
+        }
+
+        table, th, td {
+            border: 0;
+        }
+
+        .even {
+            background-color: #f0f0f0;
+        }
+
+        .value-editor {
+            margin: 2px;
+            padding: 2px;
+        }
+
+        .switch-button {
+            margin: 2px;
+            padding: 2px;
+            border-radius: 4px;
+            background: #cfcfcf;
+            border: 0;
+        }
+
+        .icon {
+            width: 24px;
+            height: 24px;
+        }
+
+        .hidden {
+            display: none;
+        }
+
+        .bold {
+            font-weight: bold;
+        }
+    </style>
+</head>
+<body>
+    <input class="search" type="text" placeholder="Search for a preference ..." aria-label="Search preferences" oninput="populate(this.value)">
+    <table></table>
+    <script>
+        // Helper to create a TD with class and optional text
+        function td(cls, text) {
+            const cell = document.createElement('td');
+            if (cls) cell.className = cls;
+            if (text !== undefined) cell.textContent = text;
+            return cell;
+        }
+
+        class PrefBoolRow extends HTMLTableRowElement {
+            connectedCallback() {
+                // Attributes are provided via data-*
+                const pref = this.getAttribute('data-pref');
+                const defaultValue = this.getAttribute('data-default') === 'true';
+                let value = this.getAttribute('data-value') === 'true';
+
+                const nameCell = td('pref-name', pref);
+                const valueCell = td('value', String(value));
+                valueCell.id = `value-${pref}`;
+                if (value !== defaultValue) valueCell.classList.add('bold');
+
+                const switcher = td('switcher');
+                const btn = document.createElement('button');
+                btn.className = 'switch-button';
+                btn.innerHTML = `
+                    <svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke-width="1.5" stroke="currentColor" class="icon">
+                        <path stroke-linecap="round" stroke-linejoin="round" d="M7.5 21 3 16.5m0 0L7.5 12M3 16.5h13.5m0-13.5L21 7.5m0 0L16.5 12M21 7.5H7.5" />
+                    </svg>`;
+                btn.onclick = () => {
+                    value = !value;
+                    if (value === defaultValue) {
```

Comment:
Nit: can use toggle to do this in one line: `valueCell.classList.toggle('bold', value !== defaultValue)`